File: calendario-spring/src/main/java/br/mil/eb/decex/calendario_spring/enumerado/PostoGraduacao.java
package br.mil.eb.decex.calendario_spring.enumerado;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum PostoGraduacao {

    GEN_EXERCITO("Gen Ex", 1),
	GEN_DIVISAO("Gen Div", 2),
	GEN_BRIGADA("Gen Bda", 3),
	CORONEL("Cel", 4),
	TEN_CORONEL("Ten Cel", 5),
	MAJOR("Maj", 6),
	CAPITAO("Cap", 7),
	PRI_TENENTE("1º Ten", 8),
	SEG_TENENTE("2º Ten", 9),
	ASP("Asp", 10),
	SUBTENENTE("ST", 11),
	PRI_SARGENTO("1º SGT", 12),
	SEG_SARGENTO("2º SGT", 13),
	TER_SARGENTO("3º SGT", 14),
	CABO("Cabo", 15),
	SOLDADO("Soldado", 16),
	FUNC_CIV("Funcionário Civil", 17);
	
	private String value;
    private int ordem;

    private PostoGraduacao(String value, int ordem) {
        this.value = value;
        this.ordem = ordem;
    }

    @JsonValue
    public String getValue() {
        return value;
    }

    public int getOrdem() {
        return ordem;
    }

    @JsonCreator
    public static PostoGraduacao fromValue(String value) {
        for (PostoGraduacao posto : PostoGraduacao.values()) {
            if (posto.getValue().equalsIgnoreCase(value)) {
                return posto;
            }
        }
        throw new IllegalArgumentException("Unknown enum value: " + value);
    }

    @Override
    public String toString() {        
        return value;
    }
	
}

File: calendario-angular/src/app/enums/PostoGraduacao/PostoGraduacao.ts

export enum PostoGraduacao {
  GEN_EXERCITO,
	GEN_DIVISAO,
	GEN_BRIGADA,
	CORONEL,
	TEN_CORONEL,
	MAJOR,
	CAPITAO,
	PRI_TENENTE,
	SEG_TENENTE,
	ASP,
	SUBTENENTE,
	PRI_SARGENTO,
	SEG_SARGENTO,
	TER_SARGENTO,
	CABO,
	SOLDADO,
	FUNC_CIV
}
export const PostoGraduacaoList = [
  { value: PostoGraduacao.GEN_EXERCITO, viewValue: 'Gen Ex' , imageUrl: 'assets/images/gen_ex.png' },
  { value: PostoGraduacao.GEN_DIVISAO, viewValue: 'Gen Div', imageUrl: 'assets/images/gen_div.png' },
  { value: PostoGraduacao.GEN_BRIGADA, viewValue: 'Gen Bda', imageUrl: 'assets/images/gen_bda.png' },
  { value: PostoGraduacao.CORONEL, viewValue: 'Cel', imageUrl: 'assets/images/cel.png' },
  { value: PostoGraduacao.TEN_CORONEL, viewValue: 'Ten Cel', imageUrl: 'assets/images/ten_cel.png' },
  { value: PostoGraduacao.MAJOR, viewValue: 'Maj', imageUrl: 'assets/images/maj.png'  },
  { value: PostoGraduacao.CAPITAO, viewValue: 'Cap', imageUrl: 'assets/images/cap.png' },
  { value: PostoGraduacao.PRI_TENENTE, viewValue: '1º Ten', imageUrl: 'assets/images/1_ten.png' },
  { value: PostoGraduacao.SEG_TENENTE, viewValue: '2º Ten', imageUrl: 'assets/images/2_ten.png' },
  { value: PostoGraduacao.ASP, viewValue: 'Asp', imageUrl: 'assets/images/asp.png' },
  { value: PostoGraduacao.SUBTENENTE, viewValue: 'ST', imageUrl: 'assets/images/st.png' },
  { value: PostoGraduacao.PRI_SARGENTO, viewValue: '1º SGT', imageUrl: 'assets/images/1_sgt.png' },
  { value: PostoGraduacao.SEG_SARGENTO, viewValue: '2º SGT', imageUrl: 'assets/images/2sgt.png'  },
  { value: PostoGraduacao.TER_SARGENTO, viewValue: '3º SGT', imageUrl: 'assets/images/3_sgt.png' },
  { value: PostoGraduacao.CABO, viewValue: 'Cabo', imageUrl: 'assets/images/cabo.png'},
  { value: PostoGraduacao.SOLDADO, viewValue: 'Soldado', imageUrl: 'assets/images/soldado.png' },
  { value: PostoGraduacao.FUNC_CIV, viewValue: 'Funcionário Civíl', imageUrl: 'assets/images/func_civ.png' },
];

const postos = Object.values(PostoGraduacao);
